### PR TITLE
allow users to edit post

### LINKED ISSUE #85: Edit Post in Post.js
One can edit their post's description, need frontend for that.

### PR DESCRIPTION
**Pull request by suhanichawla**
User can click on pencil icon in front of post description to edit it. User can click on check icon to complete editing. Relevant error and success toasts are displayed.
Attaching screenshots for reference. 
Resolves #85 

Editable description
Screenshot (298)

Toast displayed
Screenshot (299)

Edit icon
Screenshot (300)_LI

### CODE REVIEW

**Comment by aitikgupta** [comment]
All checks fail to start because the `greetings bot` itself fails due to a long-known issue https://github.com/actions/first-interaction/issues/10
The best way to deal with this would be to remove the bot, since we definitely need the linter actions to do their job.

Ping @bodhisha @DebugAgrawal to make this executive decision.

File: backend/app/server/controllers/post.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ async def add_post(email: str, post_data: dict) -> dict:
         encoded_image = await encode_image(rgb_image, info)
 
         # upload the image
-        with tempfile.NamedTemporaryFile(suffix=".png") as tmp:
+        with tempfile.NamedTemporaryFile(delete=False,suffix=".png") as tmp:
```
Comment (aitikgupta) [inline backend/app/server/controllers/post.py]:
```suggestion
        with tempfile.NamedTemporaryFile(delete=False, suffix=".png") as tmp:
```

File: backend/app/server/controllers/post.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ async def update_post(email: str, post_id: str, data: dict):
     post = await posts_collection.find_one({"_id": ObjectId(post_id)})
     if post and user["_id"] == post["user_id"]:
         updated_post = await posts_collection.update_one(
-            {"_id": ObjectId(id)}, {"$set": data}
+            {"_id": ObjectId(post_id)}, {"$set": data}
```
Comment (aitikgupta) [inline backend/app/server/controllers/post.py:119]: ahaha thanks for this^

File: backend/app/server/models/post.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ class Config:
 
 class UpdatePostModel(BaseModel):
     description: Optional[str]
+    post_id: str
```
Comment (aitikgupta) [inline backend/app/server/models/post.py]: Python suggests optional parameters to exist after the required parameters.
(example: `def func(req_param, optional_param = None)`)
```suggestion
    post_id: str
    description: Optional[str]
```

File: backend/app/server/routes/post.py [hunk 4]
```diff
@@ -37,10 +37,10 @@ async def delete_post_data(post_id: str = Body(..., embed=True), current_user=De
 
 
 @router.patch("/update", response_description="Update post")
-async def update_post_data(post_id: str, updated_post: UpdatePostModel = Body(...), current_user=Depends(auth_handler.auth_wrapper)):
+async def update_post_data(updated_post: UpdatePostModel = Body(...), current_user=Depends(auth_handler.auth_wrapper)):
     updated_post = jsonable_encoder(updated_post)
-    new_post = await update_post(current_user, post_id, updated_post)
-    return ResponseModel(new_post, "Post updated successfully.")
+    new_post = await update_post(current_user, updated_post["post_id"], updated_post)
+    return ResponseModel(new_post,"Post updated successfully.")
```
Comment (aitikgupta) [inline backend/app/server/routes/post.py]: I'm basically doing a Flake8 manual check atm. xD
```suggestion
    return ResponseModel(new_post, "Post updated successfully.")
```

File: frontend/package-lock.json [hunk 5]
```diff
@@ -3384,8 +3384,7 @@
     "binary-extensions": {
```
Comment (aitikgupta) [inline frontend/package-lock.json]: Do we need to make changes to `package-lock`?
We're not adding/removing a dependency in this PR are we?

I believe this is a mismatch between user's preinstalled packages and the repo's. (ping @bodhisha that's what I meant when I said package-lock is a mess xD)

**Review by aitikgupta — CHANGES_REQUESTED** [review]
In general, amazingg work @suhanichawla 🎉 
We don't have linter checks running on this PR atm, reviewed it anyway.

(you get browny points for fixing the FastAPI endpoint)

Netlify deployed URL: https://607805c2db109e0007a9ce8a--sociomark.netlify.app/
(it doesn't show up here because of the failing checks)

**Review by bodhisha — CHANGES_REQUESTED** [review]
@suhanichawla  Thanks for checking out the issues. PR looks good..!🎉 A small suggestion from my part, currently the description is editable by every user in the system (though not possible from the backend). So If you could show the `edit icon` only to the `post owners`, it would be cool..

**Comment by suhanichawla** [comment]
> @suhanichawla Thanks for checking out the issues. PR looks good..!🎉 A small suggestion from my part, currently the description is editable by every user in the system (though not possible from the backend). So If you could show the `edit icon` only to the `post owners`, it would be cool..

Completely slipped my mind, fixed this now, thanks!

**Comment by suhanichawla** [comment]
> In general, amazingg work @suhanichawla 🎉
> We don't have linter checks running on this PR atm, reviewed it anyway.
> 
> (you get browny points for fixing the FastAPI endpoint)
> 
> Netlify deployed URL: https://6077ecb8193d2e000830a477--sociomark.netlify.app/
> (it doesn't show up here because of the failing checks)

Committed all the indentation fixes

**Comment by aitikgupta** [comment]
Good news: Greetings bot is working fine now, via #120 

Bad news: Why are linters not running their checks? :/

**Comment by aitikgupta** [comment]
@suhanichawla you're our first PR which isn't from one of the authors, and which is why we're running into so many issues with those actions.
Just wanted to thank you for your patience, we'll try to fix those soon. :)

**Comment by aitikgupta** [comment]
@suhanichawla can you rebase to main?
Linters oughta be running now. (also the merge conflict)

**Review by aitikgupta — APPROVED** [review]
This is good to go! 🎉

**Review by bodhisha — APPROVED** [review]

**Comment by aitikgupta** [comment]
@suhanichawla thanks for your contribution! (and for the patience handling the linters :p)

**Pull request merged**